File: src/main/java/org/vitaliistf/todo/entity/Category.java
package org.vitaliistf.todo.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Collection;

@Entity
@Data
public class Category {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id")
    private Long id;

    @Basic
    @Column(name = "title")
    private String title;

    @Basic
    @Column(name = "completed_count")
    private Long completedCount;

    @Basic
    @Column(name = "uncompleted_count")
    private Long uncompletedCount;

    @OneToMany(mappedBy = "categoryByCategoryId")
    private Collection<Task> tasksById;
}

File: frontend/src/app/models/category.ts
export class Category {
  id: number;
  title: string;
  completedCount?: number;
  uncompletedCount?: number;


  constructor(id: number, title: string, completedCount?: number, uncompletedCount?: number) {
    this.id = id;
    this.title = title;

    this.completedCount = completedCount ? completedCount : 0;

    this.uncompletedCount = uncompletedCount ? uncompletedCount : 0;
  }
}
